frontend/src/modules/dataset/dataset-browser.tsx

```diff
@@ -129,94 +125,70 @@ export function DatasetBrowser (props: DatasetBrowserProps) {
-                    // DatasetBrowser invariant requires state.datasetContext can never be undefined with this displayMode
-                    if (state.datasetContext) {
-                        fetchDatasetResources(state.datasetContext);
+                    if (shouldFetch && datasetContext) {
+                        payload.isLoading = true;
+                        if (keepItems) {
+                            fetchDatasetResources(datasetContext, state.nextToken, state.items);
+                        } else {
+                            fetchDatasetResources(datasetContext);
+                        }
                     }
                     break;
                 case DatasetBrowserDisplayMode.Dataset:
-                    fetchDatasets();
+                    if (shouldFetch && datasetContext.type) {
+                        payload.isLoading = true;
+                        fetchDatasets(datasetContext.type);
+                    }
                     break;
                 default:
                     payload.items = Object.keys(DatasetType).map((key) => ({ name: upperFirst(DatasetType[key]), type: DatasetType[key] }));
-                    payload.isLoading = false;
             }
         }
 
         setState({
             type: DatasetActionType.State,
             payload
         });
-    }, [state.datasetContext, state.manageMode, fetchDatasetResources, fetchDatasets, displayMode, state.refresh]);
-
-    const pageSize = 10;
-    // recalculate pages if filteredItems or pageSize changes
-    useEffect(() => {
-        setState({
-            type: DatasetActionType.Pagination,
-            payload: {
-                pagesCount: Math.max(1, Math.ceil(state.filter.filteredItems.length / pageSize)),
-            }
-        });
-    }, [state.filter.filteredItems, pageSize]);
+    }, [fetchDatasetResources, fetchDatasets, manageMode, state.datasetContext, state.filteringText, state.items, state.nextToken]);
 
-    // if pagination changes make sure currentPageIndex isn't greater than pageCount
+    const [firstLoad, setFirstLoad] = useState(true);
     useEffect(() => {
-        if (state.pagination.currentPageIndex > state.pagination.pagesCount) {
-            setState({
-                type: DatasetActionType.Pagination,
-                payload: { currentPageIndex: state.pagination.pagesCount }
-            });
+        if (firstLoad) {
+            setFirstLoad(false);
+            updateDatasetContext(state.datasetContext || {}, '', false);
         }
-    }, [state.pagination]);
+    }, [firstLoad, state.datasetContext, updateDatasetContext]);
 
-    // if items or filteringText changes recompute filteredItems
-    useEffect(() => {
+    const filteredItems = useMemo(() => state.items.filter((item) => {
+        return !!item.name?.includes(isCreating ? state.filteringText : resourceForPath(state.filteringText));
+    }), [state.items, state.filteringText, isCreating]);
+
+    const pageSize = 10;
```

Comment:
Are we still supporting page size customization with the dataset tables?

Reply:
We aren't right now, we talked about this a few weeks ago and agreed to just set a reasonable default. I have a branch where I am updating the table component so the DatasetBrowser can use it but I need to rebase that with the changes to Dataset we've since made and the autorefresh stuff that was added recently.